Source/WebCore/css/ShorthandSerializer.cpp

```diff
@@ -1315,6 +1319,35 @@ String ShorthandSerializer::serializeTextWrap() const
     return makeString(nameLiteral(mode), ' ', nameLiteral(style));
 }
 
+String ShorthandSerializer::serializeSingleAnimationRange(CSSValue& value, TimelineRangeStartOrEnd startOrEnd, CSSValueID startValueID) const
+{
+    if (auto* pair = dynamicDowncast<CSSValuePair>(value)) {
+        if (SingleTimelineRange::isDefault(downcast<CSSPrimitiveValue>(pair->second()), TimelineRangeStartOrEnd::Start) && (startOrEnd == TimelineRangeStartOrEnd::Start || pair->first().valueID() != startValueID))
+            return nameLiteral(pair->first().valueID());
+        return serializeLonghandValue(startOrEnd == TimelineRangeStartOrEnd::Start ? 0 : 1);
+    }
+    if (auto* primitiveValue = dynamicDowncast<CSSPrimitiveValue>(value)) {
+        if (SingleTimelineRange::isOffsetValue(*primitiveValue))
+            return serializeLonghandValue(startOrEnd == TimelineRangeStartOrEnd::Start ? 0 : 1);
+        if (startOrEnd == TimelineRangeStartOrEnd::Start || (primitiveValue->valueID() != CSSValueNormal && primitiveValue->valueID() != startValueID))
+            return nameLiteral(primitiveValue->valueID());
+    }
+    return emptyString();
+}
+
+String ShorthandSerializer::serializeAnimationRange() const
+{
+    auto& start = longhandValue(0);
+    auto startID = is<CSSValuePair>(start) ? downcast<CSSValuePair>(start).first().valueID() : start.valueID();
```

Comment:
Similarly here, the common pattern is:
```suggestion
    RefPtr startPair = dynamicDowncast<CSSValuePair>(start);
    auto startID = startPair ? startPair.first().valueID() : start.valueID();
```

Reply:
Fixed.